app_test.go

```diff
@@ -860,45 +861,60 @@ func Test_App_ShutdownWithContext(t *testing.T) {
 	t.Parallel()
 
 	app := New()
+	var shutdownHookCalled atomic.Int32
+	app.Hooks().OnShutdown(func() error {
+		shutdownHookCalled.Store(1)
+		return nil
+	})
+
 	app.Get("/", func(ctx Ctx) error {
 		time.Sleep(5 * time.Second)
 		return ctx.SendString("body")
 	})
 
 	ln := fasthttputil.NewInmemoryListener()
 
+	serverErr := make(chan error, 1)
 	go func() {
-		err := app.Listener(ln)
-		assert.NoError(t, err)
+		serverErr <- app.Listener(ln)
 	}()
 
-	time.Sleep(1 * time.Second)
+	time.Sleep(100 * time.Millisecond)
 
+	clientDone := make(chan struct{})
 	go func() {
 		conn, err := ln.Dial()
 		assert.NoError(t, err)
-
-		_, err = conn.Write([]byte("GET / HTTP/1.1\r\nHost: google.com\r\n\r\n"))
+		_, err = conn.Write([]byte("GET / HTTP/1.1\r\nHost: example.com\r\n\r\n"))
 		assert.NoError(t, err)
+		close(clientDone)
 	}()
 
-	time.Sleep(1 * time.Second)
+	<-clientDone
+	// Sleep to ensure the server has started processing the request
+	time.Sleep(100 * time.Millisecond)
 
-	shutdownErr := make(chan error)
+	shutdownErr := make(chan error, 1)
 	go func() {
 		ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
 		defer cancel()
 		shutdownErr <- app.ShutdownWithContext(ctx)
 	}()
 
 	select {
-	case <-time.After(5 * time.Second):
-		t.Fatal("idle connections not closed on shutdown")
+	case <-time.After(2 * time.Second):
+		t.Fatal("shutdown did not complete in time")
```

Comment:
Is it "in time" or "on time"

Reply:
I prefer to use it `in time`, because timely means to complete a certain goal within a temporary interval, while `on time` means to complete a certain goal at a certain moment. Such as, I will have breakfast at 8 o'clock tomorrow